### DISCUSSION TITLE
Disabling Captcha Phone Authentication for E2E tests

### DISCUSSION THREAD

**Discussion by jguerena15** [firebase/firebase-ios-sdk#14146] [Q&A] [answered]
When running automated tests (iOS Firebase 10 sdk), even when using predefined test numbers, we still frequently run into captchas which prevents us from running E2E tests. I'm guessing it's due to login frequency that triggers it. During the normal login flow, most users don't see the captcha so auth is setup correctly. My question is, is there a way to bypass captcha completely for E2E tests, particularly for test numbers that we've entered? If not, are there some other common solutions to addressing this problem? 

Greatly appreciate any feedback, thank you!

**Comment by paulb777** [comment]
See https://firebase.google.com/docs/auth/ios/phone-auth

**Reply by jguerena15** [reply]
Somehow missed the part on integration testing on previous reads, thanks!

**Reply by jguerena15** [reply, marked as answer]
https://firebase.google.com/docs/auth/ios/phone-auth#integration-testing
